File: app/src/main/java/com/spidroid/starry/viewmodels/UserViewModel.java
package com.spidroid.starry.viewmodels;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;
import com.spidroid.starry.models.UserModel;

public class UserViewModel extends ViewModel {
    private final MutableLiveData<UserModel> userData = new MutableLiveData<>();

    public void clearUserData() {
        userData.setValue(null);
    }

    public LiveData<UserModel> getUserData() {
        return userData;
    }
}

File: app/src/main/java/com/spidroid/starry/viewmodels/UserViewModel.kt
package com.spidroid.starry.viewmodels



import androidx.lifecycle.LiveData

import androidx.lifecycle.MutableLiveData

import androidx.lifecycle.ViewModel

import com.spidroid.starry.models.UserModel



class UserViewModel : ViewModel() {

    private val userData = MutableLiveData<UserModel?>()



    fun clearUserData() {

        userData.setValue(null)

    }



    fun getUserData(): LiveData<UserModel?> {

        return userData

    }

}